src/components/common/YourCompany/CorrectName/CorrectNameRequest.vue

```diff
@@ -259,32 +263,21 @@ export default class CorrectNameRequest extends Mixins(CommonMixin, NameRequestM
   }
 
   /**
-   * Parse and Set the Name Request date to Store.
-   * @param nr The name request data
+   * Stores the new Name Request.
+   * @param nameRequest the name request object
    */
-  parseNameRequest (nr: NrResponseIF): void {
-    const nrCorrection: NrCorrectionIF = {
-      legalType: nr.legalType,
-      nrNumber: this.nameRequestNumber,
-      legalName: this.getNrApprovedName(nr) || '',
-      expiry: nr.expirationDate,
-      status: nr.state,
-      requestType: nr.request_action_cd,
-      applicant: {
-        fullName: this.formatFullName(nr.applicants),
-        fullAddress: this.formatFullAddress(nr.applicants),
-        phoneNumber: nr.applicants.phoneNumber,
-        emailAddress: nr.applicants.emailAddress
-      } as NameRequestApplicantIF
-    }
-
-    // set the new correction NR data
-    this.setNameRequest({ ...this.getNameRequest, ...nrCorrection })
+  storeNameRequest (nameRequest: NameRequestIF): void {
+    // overwrite the existing NR data with the new NR data
+    this.setNameRequest({
```

Comment:
Love this!

Reply:
Thanks.

An interesting thing here is that we save all properties from the original NR into the filing JSON, so that we access them when resuming (for example, applicant data). You might realize that we _don't_ reload the NR when the draft is resumed (as we do in Create UI). (It might be better if we did -- a problem for another day.) But there are still 2 checks that the NR is valid -- when File and Pay is clicked, and the by the Legal API. FYI.

Reply:
Makes sense. I guess as of right now, we're OK with not reloading the NR but maybe down the road, if the need rises, we can have that in place.